vaadin-radio-button.html

```diff
@@ -17,33 +17,169 @@
 -->
 
 <link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../polymer/lib/mixins/gesture-event-listeners.html">
+<link rel="import" href="../vaadin-themable-mixin/vaadin-themable-mixin.html">
+<link rel="import" href="../vaadin-control-state-mixin/vaadin-control-state-mixin.html">
 
-<dom-module id="vaadin-radio-button">
+<dom-module id="vaadin-radio-default-theme" theme-for="vaadin-radio-button">
   <template>
     <style>
       :host {
         display: inline-block;
+        outline: none;
+      }
+
+      [part="wrapper"] {
+        display: inline-flex;
+        align-items: center;
+        outline: none;
+      }
+
+      :host([focus-ring]) {
+        box-shadow: 0 0 2px 2px Highlight;
+      }
+
+      [part="native-radio"] {
+        margin-right: 0;
+      }
+
+      [part="label"] {
+        padding: 0 5px;
+        outline: none;
+      }
+
+      :host([disabled]) [part="label"] {
+        opacity: 0.5;
       }
     </style>
-    <slot></slot>
+  </template>
+</dom-module>
+
+<dom-module id="vaadin-radio-button">
+  <template>
+
+    <label part="wrapper" id="label">
+      <input
+        id="nativeRadio"
+        type="radio"
+        part="native-radio"
+        checked="{{checked::change}}"
+        disabled$="[[disabled]]"
+        role="presentation"
+        tabindex="-1">
+
+      <span part="radio"></span>
+
+      <span part="label">
+        <slot></slot>
+      </span>
+    </label>
+
   </template>
 
   <script>
     if (!Polymer.Element) {
-      throw new Error(`Unexpected Polymer version ${Polymer.version} is used, expected v2.0.0 or later.`);
+      throw new Error('Unexpected Polymer version ${Polymer.version} is used, expected v2.0.0 or later.');
     }
 
-    class VaadinRadioButton extends Polymer.Element {
-      static get is() {
-        return 'vaadin-radio-button';
-      }
+    (function() {
+
+      class RadioElement extends Vaadin.ControlStateMixin(Vaadin.ThemableMixin(Polymer.GestureEventListeners(Polymer.Element))) {
+        static get is() {
+          return 'vaadin-radio-button';
+        }
+
+        static get properties() {
+          return {
+
+            checked: {
+              type: Boolean,
+              value: false,
+              notify: true,
+              observer: '_checkedChanged',
+              reflectToAttribute: true
+            },
+
+            name: {
+              type: String
+            },
+
+            value: {
+              type: String,
+              reflectToAttribute: true
+            },
+
+            disabled: {
+              type: Boolean,
+              reflectToAttribute: true
+            }
+
+          };
+        }
+
+        ready() {
+
+          super.ready();
+
+          this.setAttribute('role', 'radio');
+
+          this._addActiveListeners();
+
+          if(!this.value)
```

Comment:
Is this `if` syntax allowed by code style?
And another question: what about DOM sanitizing? Passing `innerHTML` like this might be not secure for submitting forms AFAIK.